src/BlobStore.hs

```diff
@@ -56,62 +59,54 @@ import Test.QuickCheck.Instances.ByteString ()
 blobNameLength :: Int
 blobNameLength = 32
 
-newtype Password =
-  Pass BS.ByteString
-  deriving (Show)
+newtype Password = Pass BS.ByteString deriving (Show)
 
 -- HMAC specifies that keys less than the key length are null-padded up
 -- to the key length (longer keys are hashed).  This means that keys that
 -- are both less than the key length and differ only by trailing \NULs are
 -- equivalent HMAC keys.  We must surface this minor quirk here so that when
 -- we ask quickcheck for two different passwords, they actually differ.
 instance Eq Password where
-  (Pass a) == (Pass b)
+  Pass a == Pass b
     | BS.length a <= blobNameLength && BS.length b <= blobNameLength =
-      nullPad a == nullPad b
+        nullPad a == nullPad b
     | otherwise = a == b
     where
-      nullPad s = BS.append s (BS.replicate (blobNameLength - BS.length s) 0)
+      nullPad s = BS.append s $ BS.replicate (blobNameLength - BS.length s) 0
 
 instance Arbitrary Password where
   arbitrary = Pass <$> arbitrary
 
-newtype ExtantBlobName =
-  ExtantBlob BS.ByteString
-  deriving (Eq, Ord)
+newtype ExtantBlobName = ExtantBlob BS.ByteString deriving (Eq, Ord)
 
-blobName ::
-     MonadResource m => Password -> ConduitT BS.ByteString o m BS.ByteString
-blobName (Pass password) = do
-  digest <- sinkHMAC password
-  pure $ BA.convert (digest :: HMAC Crypto.Hash.Algorithms.SHA256)
+blobName
+  :: MonadResource m
+  => Password -> ConduitT BS.ByteString o m BS.ByteString
+blobName (Pass password) = BA.convert @(HMAC SHA256) <$> sinkHMAC password
```

Comment:
`TypeApplications` is really nice. I strongly advocate using it instead of inline type annotations. In this case you'll notice that `BA.convert` is a function, and if I'd used a type annotation, I would have had to provide the type for the function. `TypeApplications` like function application, but for types, so you just provide the type for each type variable in the order they appear.